icu4c/source/common/brkiter.cpp

```diff
@@ -437,6 +437,14 @@ BreakIterator::makeInstance(const Locale& loc, int32_t kind, UErrorCode& status)
                 uprv_strcat(lbType, "_");
                 uprv_strcat(lbType, lbKeyValue);
             }
+            if (strcmp(lbType, "line") == 0) {
+                char lwKeyValue[kKeyValueLenMax] = { 0 };
```

Comment:
Could you just reuse the lbKeyValue buffer  here? the content in that buffer is already no longer need after the if block above, right? you can reuse it here. Of course, you better rename that from lbKeyValue to just keyValue .
or you can rewrite whole block to
```
case UBRK_LINE:
        {
            UTRACE_ENTRY(UTRACE_UBRK_CREATE_LINE);
            uprv_strcpy(lbType, "line");
            UErrorCode kvStatus = U_ZERO_ERROR;
            CharString value;
            CharStringByteSink valueSink(&value);
            loc.getKeywordValue("lb", valueSink, kvStatus);
            if (U_SUCCESS(kvStatus) && (value == "strict" || value == "normal" || value == "loose")) {
                uprv_strcat(lbType, "_");
                uprv_strcat(lbType, value.data());
            } else {
               // lbType is still "line"
                value.clear();
                loc.getKeywordValue("lw", valueSink, kvStatus);
                if (U_SUCCESS(kvStatus) && value == "phrase") {
                    uprv_strcat(lbType, "_");
                    uprv_strcat(lbType, value.data());
                }
            }
            result = BreakIterator::buildInstance(loc, lbType, status);

            UTRACE_DATA1(UTRACE_INFO, "lb=%s", lbKeyValue);
            UTRACE_EXIT_STATUS(status);
        }
        break;
```

Reply:
Thanks. Resolved